### ISSUE TITLE
Add a way to configure simple topologySpreadConstraints and PodDisruptionBudget

### ISSUE THREAD

**Issue by umezawatakeshi** [VictoriaMetrics/operator#191]
To deploy HA monitoring cluster, it is important to configure appropriate topologySpreadConstraints (tSC) and PodDisruptionBudget (PDB). If the users want to configure simple tSCs and PDBs just covers the Pods created from the operator's custom resources, the users have to write same label selectors as the operator creates.

**Suggestion:**

- When labelSelector in tSC is not specified (= nil), the operator fills the selector with .spec.selector in the resulting Deployment or StatefulSet. Specifying nil for the labelSelector is meaningless, so I think that it does not cause a problem to use nil as a special purpose.
- Add podDisruptionBudget parameter and the operator creates PDB from it.

For example, the operator creates the following Deployment and PDB from the following VMAgent. (note that many fields which are not concerned by this suggestion are omitted)

```yaml
apiVersion: operator.victoriametrics.com/v1beta1
kind: VMAgent
metadata:
  name: example
spec:
  replicaCount: 3
  topologySpreadConstraints:
  - maxSkew: 1
    topologyKey: kubernetes.io/hostname
    whenUnsatisfiable: DoNotSchedule
    labelSelector:
      matchLabels:
        app.kubernetes.io/name: vmagent
  - maxSkew: 1
    topologyKey: kubernetes.io/os
    whenUnsatisfiable: ScheduleAnyway
    # labelSelector is not specified
  podDisruptionBudget:
    minAvailable: 2
```

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: vmagent-example
  namespace: default
spec:
  replicas: 3
  selector:
    matchLabels:
      app.kubernetes.io/component: monitoring
      app.kubernetes.io/instance: example
      app.kubernetes.io/name: vmagent
      managed-by: vm-operator
  template:
    spec:
      topologySpreadConstraints:
      - labelSelector: # specified labelSelector is just copied
          matchLabels:
            app.kubernetes.io/name: vmagent
        maxSkew: 1
        topologyKey: kubernetes.io/hostname
        whenUnsatisfiable: DoNotSchedule
      - labelSelector: # labelSelector is not specified -> copied from .spec.selector
          matchLabels:
            app.kubernetes.io/component: monitoring
            app.kubernetes.io/instance: example
            app.kubernetes.io/name: vmagent
            managed-by: vm-operator
        maxSkew: 1
        topologyKey: kubernetes.io/os
        whenUnsatisfiable: ScheduleAnyway
---
apiVersion: policy/v1beta1
kind: PodDisruptionBudget
metadata:
  name: vmagent-example
  namespace: default
spec:
  minAvailable: 2
  selector: # copied from Deployment's .spec.selector
    matchLabels:
      app.kubernetes.io/component: monitoring
      app.kubernetes.io/instance: example
      app.kubernetes.io/name: vmagent
      managed-by: vm-operator
```

I already have a POC implementation for VMAgent [here](https://github.com/umezawatakeshi/VictoriaMetrics-operator/commit/287b5dd93fea14bf32a59c5698445f6cc6e3ed9e). If you think that this design and the implementation are reasonable, I will implement the rest and send PR.

**Comment by f41gh7** [comment]
Hello, looks good to me. Quick checked implementation, only one thing is missing - removing finalizer at CRD deletion. `removeFinalizeObjByName` need to be called with PDB, https://github.com/VictoriaMetrics/operator/blob/master/controllers/factory/finalize/vmagent.go#L22 

 PR is welcome! It also nice to have such option for `VMAlert`, `VMCluster` and `VMAlertmanager`.

**Comment by umezawatakeshi** [comment]
Thank you for quick review. Please look forward to PR.

**Comment by f41gh7** [comment]
Included to 0.12.0 release
